dev/com.ibm.ws.jaxws.2.3.common/src/com/ibm/ws/jaxws/client/LibertyWebServiceClientInInterceptor.java

```diff
@@ -0,0 +1,127 @@
+/*******************************************************************************
+ * Copyright (c) 2024 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package com.ibm.ws.jaxws.client;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.Map.Entry;
+
+import org.apache.cxf.interceptor.Fault;
+import org.apache.cxf.interceptor.Interceptor;
+import org.apache.cxf.jaxb.JAXBDataBinding;
+import org.apache.cxf.message.Message;
+import org.apache.cxf.message.MessageUtils;
+import org.apache.cxf.phase.AbstractPhaseInterceptor;
+import org.apache.cxf.phase.Phase;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.ws.jaxws.internal.WebServiceConfigConstants;
+
+/**
+ *
+ */
+public class LibertyWebServiceClientInInterceptor  extends AbstractPhaseInterceptor<Message>  {
+    
+
+    private static final TraceComponent tc = Tr.register(LibertyWebServiceClientInInterceptor.class);
+    
+    /**
+     * @param serviceName
+     */
+    public LibertyWebServiceClientInInterceptor() {
+        super(Phase.RECEIVE);
+
+        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            Tr.debug(tc, "The LibertyWebServiceClientInInterceptor has been registered to the Interceptor chain.");
+        }
+    }
+
+    @Override
+    public void handleMessage(Message message) throws Fault {
+        boolean debug = TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled();
+        // Get the serviceName from the message
+        String messageServiceName = message.getExchange().getService().getName().getLocalPart();
+
+        if (debug) {
+            Tr.debug(tc, "Obtained name of the Service from the message - messageServiceName = " + messageServiceName);
+        }
+        
+        Object enableSchemaValidation;
+        
+        Object ignoreUnexpectedElements;
+        
+        // if messageServiceName != null, try to get the values from configuration using it
+        if(messageServiceName != null) {
+            // if messageServiceName != null, try to get enableSchemaValidation value from configuration, if it's == null try it to get the default configuration value
+            enableSchemaValidation = (WebServicesClientConfigHolder.getIgnoreUnexpectedElements(messageServiceName) != null) ? WebServicesClientConfigHolder.getEnableSchemaValidation(messageServiceName): WebServicesClientConfigHolder.getEnableSchemaValidation(WebServiceConfigConstants.DEFAULT_PROP);         
```

Comment:
if statement shouldn't be WebServicesClientConfigHolder.getEnableSchemaValidation(...) ?